packages/runtime/src/state/react-builder-preview.ts

```diff
@@ -287,15 +286,13 @@ export function initialize(): ThunkAction<() => void, State, unknown, Action> {
     const stopHandlingFocusEvent = startHandlingFocusEvents()
     const unlockDocumentScroll = dispatch(lockDocumentScroll())
     const stopHandlingPointerMoveEvent = dispatch(startHandlingPointerMoveEvent())
-    dispatch(setIsInBuilder(true))
```

Comment:
If this state is read-only and doesn't change anymore we probably want to get rid of the actions as well

Reply:
I dropped the commit that changes `isInBuilder`, but let's see whether we still need to add it or not. It depends on the thread down below.

Reply:
I added the commit again, somehow I cannot edit the text without that commit.

Reply:
Got it. Is the plan still to rename this to `isPreview` and rework the Text component so that it starts as readonly on changes to editable when it receives a message from the builder?

Reply:
Hmm, the commit still seems to be there on this latest version

Reply:
Not sure whether we want to rename it to `isPreview` in this PR.

For the Text component, I'm not sure how we can send the editable message from the builder reliably @migueloller . We have this code on the `RichTextPropController` in cosmos:
```ts
    constructor(api: PropControllerAPI) {
      super(api)

      // eslint-disable-next-line @typescript-eslint/no-this-alias
      const controller = this

      this.Panel = function ConnectedPanel(props) {
        const [prop, setProp] = useProp<RichTextValue>(api)

        useEffect(() => {
          let editMode = api.builder.getEditMode()

          return api.builder.subscribe(() => {
            const nextEditMode = api.builder.getEditMode()

            if (nextEditMode !== editMode) {
              editMode = nextEditMode

              controller.messageHostPropController({
                type: RichTextPropControllerMessageType.CHANGE_BUILDER_EDIT_MODE,
                editMode,
              })
            }
          })
        }, [])

        return <RichTextPanel {...props} controller={controller} value={prop} onChange={setProp} />
      }
    }
```

But because it's attached to the `Panel`, the code is only running when the `Panel` is mounted. This is a problem because for example, let's say we're already in content mode, and we want to edit another Text, that's not possible because the other Text wasn't editable because that Panel wasn't mounted yet.

I don't know where to attach the subscribe function to the controller other than on the Panel.

We could have a global store for content mode, but again, it's gonna be similar to `isInBuilder`.

Reply:
Hmm, ok. I think that we would do is subscribe when the control itself is instantiated. I guess we can take care of that another time.

So what will we be doing instead. Was there another way to solve the issue?